src/java.base/share/classes/java/net/URLDecoder.java

```diff
@@ -204,11 +205,7 @@ public static String decode(String s, Charset charset) {
 
                     while ( ((i+2) < numChars) &&
                             (c=='%')) {
-                        int v = Integer.parseInt(s, i + 1, i + 3, 16);
-                        if (v < 0)
-                            throw new IllegalArgumentException(
-                                    "URLDecoder: Illegal hex characters in escape "
-                                            + "(%) pattern - negative value");
+                        int v = HexFormat.fromHexDigits(s, i + 1, i + 3);
```

Comment:
Its worth noting that the exceptions thrown by HexFormat.fromHexDigits are different than those thrown by Integer.parseInt, as to the specific type and exception message.

Reply:
They are both NumberFormatException that is immediately handled by an enclosing catch block. This change should only affect the exception message for inputs fragments like %-F.

Reply:
I pointed it out to emphasize that exceptions and exception messages should be considered and reviewed when making a change.